ecr.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+)
+
+// for sorting setups
+// TODO: consider using global parameter 'query' to get sorted result directory from API
+// ref: https://docs.aws.amazon.com/cli/latest/reference/index.html
+type Images []ecr.ImageDetail // type alias to implement Len and Swap
+
+func (img Images) Len() int {
+	return len(img)
+}
+
+func (img Images) Swap(i, j int) {
+	img[i], img[j] = img[j], img[i]
+}
+
+type ByPushedAt struct {
+	Images
+}
+
+func (b ByPushedAt) Less(i, j int) bool {
+	ti := *b.Images[i].ImagePushedAt
+	tj := *b.Images[j].ImagePushedAt
+	return ti.Before(tj)
+}
+
+func retrieveFromECR(image string) ([]string, error) {
+	profile := os.Getenv("AWS_PROFILE")
+	if profile == "" {
+		fmt.Println("use aws default profile")
+	}
+
+	region := os.Getenv("AWS_REGION")
+	if region == "" {
+		return nil, fmt.Errorf("Error: AWS_REGION must be set")
+	}
+
+	svc := ecr.New(session.New(), &aws.Config{
+		Region: aws.String(os.Getenv("AWS_REGION")),
+	})
```

Comment:
By default, aws-sdk-go detects and uses `AWS_REGION`.
https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-the-region

```suggestion
	svc := ecr.New(session.New())
```

Reply:
OK! I removed the code for AWS variables processing in 054aedc. Thanks.